viewing_party/party.py

```diff
@@ -1,23 +1,225 @@
+        user_list.append(movies)
+    for movie in friends_list:
+        if movie not in user_list and movie not in friends_unique_movies:
+            friends_unique_movies.append(movie)
+    
+    return friends_unique_movies
+    
+
+    
+
 # ------------- WAVE 4 --------------------
-# -----------------------------------------
+# ----------------------------------------- 
+
+def get_available_recs(user_data):
+
+    movie_recs = []
+    friend_list = []
+    user_subscriptions = user_data["subscriptions"]
+    # Loop through friends watched movies to get all movies that friends watched in one liist
+    for movies1 in user_data["friends"][0]["watched"]:
+        friend_list.append(movies1)
+    for movies in user_data["friends"][1]["watched"]:
+        friend_list.append(movies)
+# Check to make sure friends movie is not in user movies, and check to make sure "host" is in user subscriptions
+    for friend in friend_list:
+        if friend not in user_data["watched"] and friend["host"] in user_subscriptions:
+            movie_recs.append(friend)
+    return movie_recs
+
+
+
+
+
+
+
 
 # -----------------------------------------
-# ------------- WAVE 5 --------------------
-# -----------------------------------------
+# ------------- WAVE 5 -------------------
+def get_new_rec_by_genre(user_data):
+
+    list_of_genres = []
+    genres = {}
+    most_watched_genre = ""
+    new_val = []
+    friends_list = []
+    new_recs = []
+
+  
+
+
+# Move user genres to a list to get most watched genre
+    for movie in user_data["watched"]:
+        list_of_genres.append(movie["genre"])
+
+# Add genre counts to a dictionary with value being how many times the genre appears in a list.
+    for genre in list_of_genres:
+        genres[genre] = genres.get(genre,0) + 1
+
+# getting highest value from dict
+    new_val = genres.values()
+    if len(new_val) > 0:
+        max_val = max(new_val)
+    for genre in genres:
+        if genres[genre] == max_val:
+            most_watched_genre = genre
+
+    for movie in user_data["friends"][0]["watched"]:
+        friends_list.append(movie)
+    for movie in user_data["friends"][1]["watched"]:
+        friends_list.append(movie)
+
+    for movies in friends_list:
+        if movies not in user_data["watched"] and movies["genre"] == most_watched_genre:
+            new_recs.append(movies)
+    return new_recs
+
+def get_rec_from_favorites(user_data):
+    user_favorites = []
+    friend_list = []
+    movie_recs = []
+    
+
+    for movies in user_data["favorites"]:
+        user_favorites.append(movies)
+    if len(user_data["friends"]) > 0:
+        for friends in user_data["friends"][0]["watched"]:
+            friend_list.append(friends)
+        for friends in user_data["friends"][1]["watched"]:
+            friend_list.append(friends)
+
+    for movies in user_favorites:
+        if movies not in friend_list:
+            movie_recs.append(movies)
+    return movie_recs

```

Comment:
👍  Nice work! We can use a guard clause here to allow the logic for building `friend_list` to be less indented. 

```python
def get_rec_from_favorites(user_data):
    user_favorites = []
    friend_list = []
    movie_recs = []

    if not user_data or not user_data["friends"]:
        return 

    for movies in user_data["favorites"]:
        user_favorites.append(movies)
        
    for friends in user_data["friends"][0]["watched"]:
        friend_list.append(friends)
    for friends in user_data["friends"][1]["watched"]:
        friend_list.append(friends)

    for movies in user_favorites:
        if movies not in friend_list:
            movie_recs.append(movies)
    return movie_recs
    
```